src/mlpack/core/tree/space_split/projection_vector.hpp

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file projection_vector.hpp
+ * @author Marcos Pividori
+ *
+ * Definition of ProjVector and AxisParallelProjVector.
+ */
+#ifndef MLPACK_CORE_TREE_SPILL_TREE_PROJECTION_VECTOR_HPP
+#define MLPACK_CORE_TREE_SPILL_TREE_PROJECTION_VECTOR_HPP
+
+#include <mlpack/core.hpp>
+#include "../bounds.hpp"
+namespace mlpack {
+namespace tree {
+
+/**
+ * AxisParallelProjVector defines an axis-parallel projection vector.
+ * We can efficiently project points, simply analyzing a specific dimension.
+ */
+class AxisParallelProjVector
+{
+  //! Dimension considered.
+  size_t dim;
+
+ public:
+  /**
+   * Create the projection vector based on the specified dimension.
+   *
+   * @param dim Dimension to be considered.
+   */
+  AxisParallelProjVector(size_t dim = 0) :
+      dim(dim)
+  {};
+
+  /**
+   * Project the given point on the projection vector.
+   *
+   * @param point Point to be projected.
+   */
+  template<typename VecType>
+  double Project(const VecType& point,
+                 typename boost::enable_if<IsVector<VecType> >::type* = 0) const
+  {
+    return point[dim];
+  };
+
+  /**
+   * Project the given hrect bound on the projection vector.
+   *
+   * @param bound Bound to be projected.
+   * @return Range of projected values.
+   */
+  template<typename MetricType>
```

Comment:
`HRectBound` also has an `ElemType` template parameter, so either we should add `ElemType` as a template parameter to this function, or to this class.  Otherwise this will not work for trees built with different `ElemType`s.